keras_nlp/samplers/top_p_sampler.py

```diff
@@ -65,17 +67,28 @@ def next(prompt, cache, index):
     def __init__(
         self,
         p=0.1,
+        sample_from_top_k=40,
         seed=None,
     ):
         super().__init__()
         self.p = p
+        self.sample_from_top_k = sample_from_top_k
         self.seed = seed
 
     def get_next_token(self, probabilities):
-        # Sort preds in descending order.
-        sorted_preds, sorted_indices = tf.math.top_k(
-            probabilities, k=tf.shape(probabilities)[1], sorted=True
-        )
+        if self.sample_from_top_k is not None:
```

Comment:
This can be simpler.

```
cutoff = tf.shape(probabilities)[1]
if self.k is not None:
    cutoff = tf.math.minimum(cutoff, self.k)
sorted_preds, sorted_indices = tf.math.top_k(probabilities, k=cutoff, sorted=True)